modules/layers/src/column-layer/column-layer.js

```diff
@@ -105,69 +120,121 @@ export default class ColumnLayer extends Layer {
+    return new ColumnGeometry({
       radius: 1,
       topCap: false,
-      bottomCap: true,
+      bottomCap: mode === FILL_MODE,
+      mode,
+      drawMode: mode === FILL_MODE ? GL.TRIANGLES : GL.LINES,
       height: 2,
       verticalAxis: 'z',
       nradial: diskResolution,
       nvertical: 1
     });
   }
 
-  _getModel(gl) {
-    return new Model(
-      gl,
-      Object.assign({}, this.getShaders(), {
-        id: this.props.id,
-        geometry: this.getGeometry(this.props.diskResolution),
-        isInstanced: true,
-        shaderCache: this.context.shaderCache
-      })
-    );
+  _getModels(gl) {
+    const {id, filled, wireframe, diskResolution} = this.props;
+    let fillModel;
+    let strokeModel;
+
+    if (filled) {
+      fillModel = new Model(
+        gl,
+        Object.assign({}, this.getShaders(), {
+          id: `${id}-${FILL_MODE}`,
+          uniforms: {isWireframe: false},
+          geometry: this.getGeometry(diskResolution, FILL_MODE),
+          isInstanced: true,
+          shaderCache: this.context.shaderCache
+        })
+      );
+    }
+    if (wireframe) {
+      strokeModel = new Model(
+        gl,
+        Object.assign({}, this.getShaders(), {
+          id: `${id}-${STROKE_MODE}`,
+          uniforms: {isWireframe: true},
+          geometry: this.getGeometry(diskResolution, STROKE_MODE),
+          isInstanced: true,
+          shaderCache: this.context.shaderCache
+        })
+      );
+    }
+
+    return {
+      models: [fillModel, strokeModel].filter(Boolean),
+      fillModel,
+      strokeModel
+    };
   }
 
   _updateVertices(vertices) {
     if (!vertices) {
       return;
     }
-
     const {diskResolution} = this.props;
     log.assert(vertices.length >= diskResolution);
 
-    const {model} = this.state;
-    const geometry = this.getGeometry(this.props.diskResolution);
-    const positions = geometry.attributes.POSITION;
-    let i = 0;
-    for (let loopIndex = 0; loopIndex < 3; loopIndex++) {
-      for (let j = 0; j <= diskResolution; j++) {
-        const p = vertices[j] || vertices[0]; // auto close loop
-        // replace x and y in geometry
-        positions.value[i++] = p[0];
-        positions.value[i++] = p[1];
-        i++;
+    const modes = [FILL_MODE, STROKE_MODE];
+
+    modes.forEach((mode, index) => {
+      const model = this.state.models[index];
+      if (model) {
+        const geometry = this.getGeometry(diskResolution, mode);
+        const positions = geometry.attributes.POSITION;
+        let i = 0;
+        for (let loopIndex = 0; loopIndex < 3; loopIndex++) {
+          for (let j = 0; j <= diskResolution; j++) {
+            const p = vertices[j] || vertices[0]; // auto close loop
+            // replace x and y in geometry
+            positions.value[i++] = p[0];
+            positions.value[i++] = p[1];
+            i++;
+          }
+        }
+        model.setProps({geometry});
       }
-    }
-    model.setProps({geometry});
+    });
```

Comment:
I would prefer to explicitly say `fillModel`, and `strokeModel`, which is more clear. 

```
const {fillModel, strokeModel} = this.state;

if (fillModel) {
  // get the geometry
  // update vertices
}

if (strokeModel) {
  // get the geometry
  // update vertices
}
```

`update vertices` could be extracted into a function.